core/src/main/java/org/apache/druid/timeline/partition/DimensionRangeShardSpec.java

```diff
@@ -145,29 +141,81 @@ private static ShardSpecLookup createLookup(List<? extends ShardSpec> shardSpecs
     return Collections.unmodifiableList(dimensions);
   }
 
-  private Range<String> getFirstDimRange()
+  /**
+   * Check if a given domain of Strings is a singleton set containing the given value
+   * @param rangeSet Domain of Strings
+   * @param val Value of String
+   * @return rangeSet == {val}
+   */
+  private boolean isRangeSetSingletonWithVal(RangeSet<String> rangeSet, String val)
   {
-    Range<String> range;
-    if (firstDimStart == null && firstDimEnd == null) {
-      range = Range.all();
-    } else if (firstDimStart == null) {
-      range = Range.atMost(firstDimEnd);
-    } else if (firstDimEnd == null) {
-      range = Range.atLeast(firstDimStart);
-    } else {
-      range = Range.closed(firstDimStart, firstDimEnd);
+    if (val == null) {
+      return false;
     }
-    return range;
+    return rangeSet.asRanges().equals(
+        Collections.singleton(Range.singleton(val))
+    );
   }
 
+  /**
+   * Pruning scenarios at dimension i: (When the Effective domain is empty)
+   *  0) query domain left-aligns with start and right-aligns with end and then deviates outside to align with neither
+   *  1) The query domain left-aligns with start until dimension i and then deviates to be strictly less than start[i]
+   *  2) The query domain right-aligns with end until dimension i and then deviates to be strictly greater than end[i]
+   *
+   * Immediate acceptance criteria at dimension i:
+   *  Effective domain is non-empty and no longer aligns with any boundary
```

Comment:
Is there one more scenario more scenario:- Effective domain is non empty, aligns throughout all dimensions, but no more dimension left. (final `return true` case)?